cmd/retester/main_test.go

```diff
@@ -1,19 +1,125 @@
 package main
 
 import (
+	"errors"
 	"os"
 	"testing"
+	"time"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/sirupsen/logrus"
+
+	flagutil "k8s.io/test-infra/prow/flagutil/config"
+
+	"github.com/openshift/ci-tools/pkg/testhelper"
+)
+
+var (
+	hour      = time.Duration(1000000000 * 3600)
+	sevenDays = 168 * hour
 )
 
 func TestGatherOptions(t *testing.T) {
-	expected := "config.yaml"
-	os.Args = []string{"cmd", "--cache-record-age=100h", "--config-file=config.yaml"}
+	defer func() {
+		logrus.StandardLogger().ExitFunc = nil
+	}()
+	var fatalErr bool
+	logrus.StandardLogger().ExitFunc = func(int) { fatalErr = true }
```

Comment:
Why do we need this?

Reply:
Yes, we need this in case we want to catch fatal error from logrus.